### PR TITLE
Spring pageable support

### PR DESCRIPTION
**Pull request by Zomzog**
#### What's this PR do/fix?

Add spring `Pageable` configuration support for #2761

#### Are there unit tests? If not how should this be manually tested?

The unit test is partial

#### Any background context you want to provide?

I've added a hack for adding description and example. I think it's useful but I can remove it.

#### What are the relevant issues?

WIthout this config, generated swagger contain invalid fields and generate noncompilable code with [openapi-generator](https://github.com/OpenAPITools/openapi-generator)

### CODE REVIEW

File: springfox-spring-integration/build.gradle [hunk 1]
```diff
@@ -16,8 +16,10 @@ dependencies {
   compile project(':springfox-spi')
   compile project(':springfox-schema').sourceSets.main.output
   compile project(':springfox-spring-web')
+  compile project(':springfox-swagger-common')
```
Comment (dilipkrish) [inline springfox-spring-integration/build.gradle]: Why do we need this dependency?

File: springfox-spring-integration/build.gradle [hunk 2]
```diff
@@ -16,8 +16,10 @@ dependencies {
...
   compile "com.athaydes.rawhttp:rawhttp-core:2.0"
   compileOnly "org.springframework.restdocs:spring-restdocs-mockmvc:2.0.3.RELEASE"
+  compileOnly "org.springframework.data:spring-data-rest-webmvc:$springDataRest"
```
Comment (dilipkrish) [inline springfox-spring-integration/build.gradle:26]: Can this be provided?

Comment (Zomzog) [reply springfox-spring-integration/build.gradle:26]: Not sure to understand what you mean by "provided".

File: springfox-spring-integration/build.gradle (hunk 1 above)
Comment (Zomzog) [reply springfox-spring-integration/build.gradle]: It's for adding annotation but maybe I can remove them.

**Comment by Zomzog** [comment]
@dilipkrish I've rebased it, I will try to answer faster this time.

**Comment by ryandanielspmc** [comment]
@Zomzog @dilipkrish One issue I see here is that this code uses the word `limit`, whereas, according to this article: https://www.baeldung.com/spring-data-web-support and this article: https://reflectoring.io/spring-boot-paging/, the parameter created by spring is called `size`.

**Comment by Zomzog** [comment]
> @Zomzog @dilipkrish One issue I see here is that this code uses the word `limit`, whereas, according to this article: https://www.baeldung.com/spring-data-web-support and this article: https://reflectoring.io/spring-boot-paging/, the parameter created by spring is called `size`.

good point, I've fixed the name.

**Pull request closed without merging**